### PR TITLE
#18424 Issue: Added functions reboot and wait_reboot

### PR DESCRIPTION
**Pull request by mrdetective**
I have made some updates to the `MachineCase` class in the `testlib.py` file. Specifically, I have added two new functions: `reboot` and `wait_reboot`. These functions now support a `timeout_sec` parameter, and I have implemented function overloading to achieve this.

Additionally, I have included a call to `self.machine.allow_restart_journal_messages()` within both of these functions. This will automatically trigger whenever either of the two functions is called.

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -2131,6 +2131,22 @@ def setup_provisioned_hosts(self, disable_preload: bool = False):
             m.execute(f"hostnamectl set-hostname {name}")
             if disable_preload:
                 self.disable_preload("packagekit", "playground", "systemd", machine=m)
+    def reboot(self,timeout_sec):
+        self.machine.allow_restart_journal_messages()
+        self.machine.reboot(timeout = timeout_sec)
+    
+    def reboot(self):
```
Comment (KKoukiou) [inline test/common/testlib.py:2138]: Remove duplicate.

File: test/common/testlib.py [hunk 2]
```diff
@@ -2131,6 +2131,22 @@ def setup_provisioned_hosts(self, disable_preload: bool = False):
...
+        self.machine.allow_restart_journal_messages()
+        self.machine.reboot()
+    
+    def wait_reboot(self, timeout_sec): 
+        self.machine.allow_restart_journal_messages()
+        self.machine.wait_reboot(timeout_sec = timeout_sec)
+
+    def wait_reboot(self):
```
Comment (KKoukiou) [inline test/common/testlib.py:2146]: Please remove duplicate.

Comment (KKoukiou) [reply test/common/testlib.py:2146]: This is great, but nothing actually uses it. Please adjust the existing tests to use the new self.wait_reboot. For example testTracer in test/verify/check-packagekit.

    Adjust the usage of m.wait_reboot in test/verify/check-packagekit
    Prepare a test virtual machine: ./test/image-prepare -q fedora-37
    Run the test: TEST_OS=fedora-37 ./test/verify/check-packagekit -vst

If this works as expected (test passes), adjust all other occurrences of m.wait_reboot in test/verify.

See original comment at https://github.com/cockpit-project/cockpit/pull/18431/files#r1122949054

**Review by KKoukiou — CHANGES_REQUESTED** [review]

**Comment by mrdetective** [comment]
Thank you for reviewing my changes. I have made the requested modifications, but I'm unsure about how to test them. Could you please provide me with some guidance or resources on how to properly test the changes?

**Comment by KKoukiou** [comment]
> Thank you for reviewing my changes. I have made the requested modifications, but I'm unsure about how to test them. Could you please provide me with some guidance or resources on how to properly test the changes?

No problem please take a look here https://github.com/cockpit-project/cockpit/blob/main/test/README.md and reach out to me on [matrix](https://github.com/cockpit-project/cockpit/wiki/About#contact-developers--stay-up-to-date) in case you need further guidance.

**Comment by mrdetective** [comment]
Based on the information you provided, it sounds like @jelly suggested making changes to the `wait_reboot()` function in all the places within test\verify. It is unclear whether the `reboot()` function needs to be modified in the same way without further context.  
And also for functions without parameters can I simply put `(timeout_sec = 0)`? 
for example,  `m.reboot()` will become `m.reboot(timeout_sec = 0)`

**Comment by jelly** [comment]
> Based on the information you provided, it sounds like @jelly suggested making changes to the `wait_reboot()` function in all the places within test\verify. It is unclear whether the `reboot()` function needs to be modified in the same way without further context. And also for functions without parameters can I simply put `(timeout_sec = 0)`? for example, `m.reboot()` will become `m.reboot(timeout_sec = 0)`

The suggestion is to wrap the Machine's reboot function in the testlib MachineCase class, which you did. And then to replace our tests which call `m.reboot()` with `self.reboot()`.

File: test/verify/check-kdump [hunk 3]
```diff
@@ -27,7 +27,7 @@ class KdumpHelpers(MachineCase):
         # all current Fedora/CentOS/RHEL images use BootLoaderSpec
         self.machine.execute("grubby --args=crashkernel=256M --update-kernel=ALL")
         self.machine.execute("mkdir -p /var/crash")
-        self.machine.reboot()
+        self.reboot()
         self.allow_restart_journal_messages()
```
Comment (KKoukiou) [inline test/verify/check-kdump:31]: No allow_restart_journal messages() call can be removed from all of the tests, when ths follows the new reboot or wait_reboot calls.

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: test/common/testlib.py [hunk 4]
```diff
@@ -2131,6 +2131,15 @@ def setup_provisioned_hosts(self, disable_preload: bool = False):
             m.execute(f"hostnamectl set-hostname {name}")
             if disable_preload:
                 self.disable_preload("packagekit", "playground", "systemd", machine=m)
+    def reboot(self, timeout_sec):
```
Comment (KKoukiou) [inline test/common/testlib.py:2134]: Needs empty line before new function

File: test/common/testlib.py [hunk 5]
```diff
@@ -2131,6 +2131,15 @@ def setup_provisioned_hosts(self, disable_preload: bool = False):
...
+        self.machine.allow_restart_journal_messages()
+        self.machine.reboot(timeout_sec = timeout_sec)
+        
+    def wait_reboot(self, timeout_sec): 
+        self.machine.allow_restart_journal_messages()
+        self.machine.wait_reboot(timeout_sec = timeout_sec)
+
```
Comment (KKoukiou) [inline test/common/testlib.py:2141]: Remove these two extra new lines.

Comment (KKoukiou) [reply test/common/testlib.py:2141]: Run ./test/static-code to see the linter errors.

File: test/common/testlib.py [hunk 6]
```diff
@@ -2131,6 +2131,15 @@ def setup_provisioned_hosts(self, disable_preload: bool = False):
             m.execute(f"hostnamectl set-hostname {name}")
             if disable_preload:
                 self.disable_preload("packagekit", "playground", "systemd", machine=m)
+    def reboot(self, timeout_sec):
+        self.machine.allow_restart_journal_messages()
+        self.machine.reboot(timeout_sec = timeout_sec)
+        
+    def wait_reboot(self, timeout_sec):
```
Comment (KKoukiou) [inline test/common/testlib.py:2138]: timeout_sec needs a default value, None.

File: test/verify/check-system-realms [hunk 7]
```diff
@@ -338,7 +338,7 @@ class CommonTests:
         b.click("#shutdown-dialog button:contains(Reboot)")
         b.switch_to_top()
         b.wait_in_text(".curtains-ct h1", "Disconnected")
-        m.wait_reboot()
```
Comment (KKoukiou) [inline test/verify/check-system-realms:341]: self.allow_restart_journal_messages() can be removed on line 344.

File: test/verify/check-system-services [hunk 8]
```diff
@@ -1545,7 +1545,7 @@ class TestTimers(MachineCase):
         b.click("#timer-save-button")
         b.wait_not_present("#timer-dialog")
         b.wait_visible(self.svc_sel('boot_timer.timer'))
-        m.reboot()
```
Comment (KKoukiou) [inline test/verify/check-system-services:1548]: self.allow_restart_journal_messages() can be removed on line 344.

File: test/verify/check-system-info [hunk 9]
```diff
@@ -746,7 +746,7 @@ machine         : 8561
                        (self.expect_smt_default and ':not(:checked)' or ':checked'))
         b.click('#cpu-mitigations-dialog Button:contains(Save and reboot)')
 
-        m.wait_reboot()
+        self.wait_reboot()
```
Comment (KKoukiou) [inline test/verify/check-system-info:749]: self.allow_restart_journal_messages() can be removed from line 988.

**Review by KKoukiou — CHANGES_REQUESTED** [review]
git grep allow_restart_journal_messages. There are a bunch of occurances still that can be removed. Check if in the same test the this is called we do call the reboot functions. If so the alow_restart_journal_messages call can be removed now safely.

**Review by KKoukiou — CHANGES_REQUESTED** [review]
All tests that are affected now fail with 'AttributeError: 'VirtMachine' object has no attribute 'allow_restart_journal_messages''

I think you need to call it like, self.allow._restart...() instead of self.machine.allow_restart() in the reboot and wait_reboot methods.

**Comment by KKoukiou** [comment]
@mrdetective this needs a rebase.

**Comment by KKoukiou** [comment]
Closing because of inactivitry. Feel free to reopen if interested.

**Pull request closed without merging**